R/logo.R

```diff
@@ -0,0 +1,45 @@
+
+#' Add Fred Hutch DaSL logo
+#'
+#' Logo placed at the bottom right corner of a ggplot
+#'
+#' @return A list with ggplot2 components : `annotation_custom()`, `coord_cartesian()`, `theme()`)
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' library(ggplot2)
+#'
+#' ggplot(diamonds[sample(nrow(diamonds), 1000), ], aes(carat, price)) +
+#' geom_point(aes(colour = color)) +
+#' scale_colour_palette_d(hutch_palette) +
+#' dasl_logo()
+#' }
+dasl_logo <- function() {
+  # Borrowed from
+  # https://www.markhw.com/blog/logos
+  # https://stackoverflow.com/q/76354700/14804653
+  list(
+    ggplot2::annotation_custom(render_png(system.file("logo", "dasl-logo.png",
+                                             package = "hutchplot"))),
+    ggplot2::coord_cartesian(clip = "off"),
+    ggplot2::theme(plot.margin = grid::unit(c(1, 1, 1.5, 1), "lines"))
+  )
+}
+
+# Render raster object
+render_png <- function(file_name) {
```

Comment:
You may need to add some arguments that allow people to adjust the logo if it doesn't fit well in the particular position for their plot.

Reply:
Good idea!